### PR TITLE
Productionise snapshot settings

### PR DESCRIPTION
**Pull request by mfw78**
# Description
This PR handles necessities to ensure QA with snapshot settings, moving towards production.

# Changes

- [x] Repository setup for devcontainers / CICD
- [x] Enforces minimum vCOW/COW holdings to prevent spam on governance proposals
- [x] Enforces a minimum participation of governance token holders (quorum)
- [x] Automatically uploads the settings to Pinata

## How to test
1. `yarn test`, or
2. CI/CD

### CODE REVIEW

**Comment by mfw78** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by mfw78** [comment]
recheck

File: package.json [hunk 1]
```diff
@@ -24,6 +25,7 @@
     "eslint-config-prettier": "^9.0.0",
     "eslint-plugin-import": "^2.29.0",
     "eslint-plugin-prettier": "^5.0.1",
+    "ipfs-deploy": "^12.0.1",
```
Comment (fedgiac) [inline package.json:28]: Just FYI: while taking a look to this package I noticed it's [deprecated](https://www.npmjs.com/package/ipfs-deploy).  As this is the same process we use in the fronted it's ok to use it now, but we might have to change it at some point.

File: test/settings.test.ts [hunk 2]
```diff
@@ -8,4 +8,10 @@ describe("settings.json", () => {
       snapshot.utils.validateSchema(snapshot.schemas.space, settings),
     ).toBe(true);
   });
+  test("Spam protection: require 10000 COW/vCOW", () => {
+    expect(settings.validation.params.minScore == 10000).toBe(true);
+  });
+  test("Participation: require 35M COW/vCOW quorum", () => {
+    expect(settings.voting.quorum == 35000000).toBe(true);
+  });
```
Comment (fedgiac) [inline test/settings.test.ts]: Nit for readability of tests and test output:

```suggestion
  test("Spam protection: require 10000 COW/vCOW", () => {
    expect(settings.validation.params.minScore).toBe(10_000);
  });
  test("Participation: require 35M COW/vCOW quorum", () => {
    expect(settings.voting.quorum).toBe(35_000_000);
  });
```

**Review by fedgiac — APPROVED** [review]
Looks good to me. Unfortunately I can't test the dev containers (not set up) or IPFS (no Pinata keys).

File: package.json (hunk 1 above)
Comment (mfw78) [reply package.json:28]: Yeah, I was not really happy about going down this route either. I checked some github actions for IPFS upload and they seemed to fail with various problems, so elected for something stable.

File: test/settings.test.ts (hunk 2 above)
Comment (mfw78) [reply test/settings.test.ts]: Nice point!

**Pull request merged**